src/store/modules/web3.ts

```diff
@@ -83,20 +83,26 @@ const actions = {
         auth.provider.value.removeAllListeners();
       if (auth.provider.value.on) {
         auth.provider.value.on('chainChanged', async chainId => {
+          commit('setLoading', true);
+
           await dispatch('account/resetAccount', null, { root: true });
           auth.web3 = new Web3Provider(auth.provider.value);
+
           commit('setNetwork', parseInt(formatUnits(chainId, 0)));
           dispatch('account/getBalances', null, { root: true });
           dispatch('account/getAllowances', null, { root: true });
+          commit('setLoading', false);
         });
         auth.provider.value.on('accountsChanged', async accounts => {
           if (accounts.length !== 0) {
+            commit('setLoading', true);
+
             await dispatch('account/resetAccount', null, { root: true });
             auth.web3 = new Web3Provider(auth.provider.value);
             commit('setAccount', accounts[0]);
-            await dispatch('loadProvider');
```

Comment:
You're right to try and remove this, but are you sure it won't have any unintended consequences?

Reply:
It will simply re-register everything, there will be none